pkg/cluster/spec/validate_test.go

```diff
@@ -428,12 +428,12 @@ tidb_servers:
 	c.Assert(ok, IsTrue)
 	c.Assert(suggestion, Equals, `The port you specified in the topology file is:
   Port:      9100
-  Component: tidb 172.16.5.138
+  Component: monitor 172.16.5.138
```

Comment:
Did you tested if the conflict is between a tidb node and a pd node but not with their monitoring agent's ports?

Reply:
The 9100 is the default port of the monitoring agent.
And I debug the code step by step to know that the conflict is monitoring agent after seeing the misleading error message https://github.com/pingcap/tiup/issues/965#issuecomment-738515276

For the test case that conflicts between tidb node and pd node but not their monitoring agent's, there is a test case between pump_server and pd in the following lines.